frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/base/BaseSlideInNewBranchBelowAction.java

```diff
@@ -0,0 +1,136 @@
+
+  @Override
+  public IEnhancedLambdaLogger log() {
+    return LOG;
+  }
+
+  @Override
+  @UIEffect
+  public void update(AnActionEvent anActionEvent) {
+    super.update(anActionEvent);
+
+    Presentation presentation = anActionEvent.getPresentation();
+    if (!presentation.isEnabledAndVisible()) {
+      return;
+    }
+
+    var branchName = getNameOfBranchUnderAction(anActionEvent);
+    var branch = branchName.flatMap(bn -> getGitMacheteBranchByName(anActionEvent, bn));
+
+    if (branch.isEmpty()) {
+      presentation.setEnabled(false);
+      presentation.setDescription(GitMacheteBundle.message("action.slide-in.description.disabled.no-parent"));
+    } else {
+      // it is supposed to be defined since "branch" is defined
+      assert branchName.isDefined() : "Branch name is undefined";
+      presentation.setDescription(GitMacheteBundle.message("action.slide-in.description", branchName.get()));
+
+      if (getCurrentBranchNameIfManaged(anActionEvent).equals(branchName)) {
+        presentation.setText(GitMacheteBundle.message("action.slide-in.text"));
+      }
+    }
+  }
+
+  @Override
+  public void actionPerformed(AnActionEvent anActionEvent) {
+    var project = getProject(anActionEvent);
+    var selectedVcsRepository = getSelectedGitRepository(anActionEvent);
+    var gitMacheteParentBranch = getNameOfBranchUnderAction(anActionEvent)
+        .flatMap(bn -> getGitMacheteBranchByName(anActionEvent, bn));
+    var hash = gitMacheteParentBranch.map(b -> b.getPointedCommit()).map(c -> c.getHash());
+    var branchLayout = getBranchLayout(anActionEvent);
+    var branchLayoutWriter = getBranchLayoutWriter(anActionEvent);
+
+    if (selectedVcsRepository.isEmpty() || gitMacheteParentBranch.isEmpty() || branchLayout.isEmpty()) {
+      return;
+    }
+
+    var branchName = createOrCheckoutNewBranch(project, selectedVcsRepository.get(), gitMacheteParentBranch.get().getName(),
+        GitMacheteBundle.message("action.slide-in.dialog.title"));
+    if (branchName == null) {
+      log().debug("Name of branch to slide in is null: most likely the action has been canceled from dialog");
+      return;
+    }
+
+    new Task.Backgroundable(project, GitMacheteBundle.message("action.slide-in.task.title")) {
+
+      @Override
+      public void run(ProgressIndicator indicator) {
+        Path macheteFilePath = getMacheteFilePath(selectedVcsRepository.get());
+        var notifier = VcsNotifier.getInstance(project);
+
+        var newBranchLayout = Try.of(() -> branchLayout.get().slideIn(gitMacheteParentBranch.get().getName(), branchName))
+            .onFailure(
+                t -> notifier.notifyError(/* title */ GitMacheteBundle.message("action.slide-in.notification.fail", branchName),
+                    getMessageOrEmpty(t)))
+            .toOption();
+
+        newBranchLayout.map(nbl -> Try.of(() -> {
+          branchLayoutWriter.write(macheteFilePath, nbl, /* backupOldLayout */ true);
+          return "ok";
```

Comment:
`Try.run` - and then this `return` might become unnecessary

Reply:
nice !